sdks/java/extensions/google-cloud-platform-core/src/main/java/org/apache/beam/sdk/util/GcsUtil.java

```diff
@@ -189,18 +168,18 @@ public static String getGlobPrefix(String globExp) {
    * @param globExp the glob expression to expand
    * @return a string with the regular expression this glob expands to
    */
-  public static String globToRegexp(String globExp) {
+  public static String wildcardToRegexp(String globExp) {
     StringBuilder dst = new StringBuilder();
-    char[] src = globExp.toCharArray();
+    char[] src = globExp.replace("**/*", "**").toCharArray();
     int i = 0;
     while (i < src.length) {
       char c = src[i++];
       switch (c) {
         case '*':
-          dst.append("[^/]*");
+          dst.append(".*");
           break;
         case '?':
-          dst.append("[^/]");
+          dst.append(".");
```

Comment:
I'm not sure this is quite the right interpretation here. If I ask for `f???.txt` I probably don't intend for `f/g/.txt` to be captured in the regex.

Thoughts?